### ISSUE TITLE
simple_trainer.py RGBA dataset

### ISSUE THREAD

**Issue by sam598** [nerfstudio-project/gsplat#372]
It looks like simple_trainer does not take into account transparency in the input images, even though it has a random_bkgd field.

Is this as simple as bringing in the 4th channel and matching the random color of the background? Or is the "[transparency carving](https://github.com/nerfstudio-project/nerfstudio/pull/2889)" technique in splatfacto more complex than this?

**Comment by liruilong940607** [comment]
If you have access of the alpha channel of the input images, there are actually two ways to use it.

- Option 1 is super simple, you could just apply an L1/L2 loss between the rendered alpha map with the ground-truth alpha channel.

- Option 2 is like the transparency carving, where you generate a random color at each iteration. Then you augment the ground-truth RGB image with this random color using the alpha channel, as well as the rendered image:
```
rng_rgb: ...

gt_rgb = gt_rgb * gt_alpha + rng_rgb * (1 - gt_apha)
pred_rgb = pred_rgb + rng_rgb * (1 - pred_alpha)

loss = L2(gt_rgb, pred_rgb)
```

Actually, for option 2, if the loss converge to zero, you would get `pred_rgb=gt_rgb * gt_alpha` and `pred_alpha=gt_apha`, which is exactly what you want. And same for the option 1.

**Comment by Ben-Mack** [comment]
@AlexMorgand A PR for this would be greatly appreciated, I think transparency bg is the most major diff that gsplat missing vs splatfacto.
